# Challenges are brief tasks designed to help you practice specific skills:

🗺️ Explore: Which of the three seventh generation consoles (Xbox 360, Playstation 3, and Nintendo Wii) had the highest total sales globally?
📊 Visualize: Create a plot visualizing the average sales for games in the most popular three genres. Differentiate between NA, EU, and global sales.
🔎 Analyze: Are some genres significantly more likely to perform better or worse in Japan than others? If so, which ones?

## PROBLEM DESCRIPTION
You are working as a data analyst for a video game retailer based in Japan. The retailer typically orders games based on sales in North America and Europe, as the games are often released later in Japan. However, they have found that North American and European sales are not always a perfect predictor of how a game will sell in Japan.

Your manager has asked you to develop a model that can predict the sales in Japan using sales in North America and Europe and other attributes such as the name of the game, the platform, the genre, and the publisher.

You will need to prepare a report that is accessible to a broad audience. It should outline your motivation, steps, findings, and conclusions.

![video_game](video_game.jpg)


In [2]:
# best selling games
-- best_selling_games
select *
FROM public.game_sales
order by public.game_sales.games_sold desc
LIMIT 10;

Error: # best selling games
-- best_selling_games
select *
FROM public.game_sales
order by public.game_sales.games_sold desc
LIMIT 10; - syntax error at or near "#"

#  BEST SELLING GAME
Based on the query results, the best-selling game was 
"Wii Sports" for the Wii, a game published by Nintendo. In addition to the query results, Nintendo appears 7 out of the 10 ranked results as the best-selling games publisher from 1985-2009. After 2009, Nintendo publishers no longer appear as the best-selling game's publisher.

In [3]:
-- critics_top_ten_years
select	
	round(public.critics_avg_year_rating.avg_critic_score, 2) as avg_critic_score,
	public.critics_avg_year_rating.year,
	public.critics_avg_year_rating.num_games,
	rank() over (order by public.critics_avg_year_rating.avg_critic_score desc) as avg_rank
from public.critics_avg_year_rating
where public.critics_avg_year_rating.num_games > 4
limit 10;

avg_critic_score  year  num_games  avg_rank
0              9.32  1998         10         1
1              9.03  2004         11         2
2              8.99  2002          9         3
3              8.93  1999         11         4
4              8.82  2001         13         5
5              8.76  2011         26         6
6              8.67  2016         13         7
7              8.66  2013         18         8
8              8.63  2008         20         9
9              8.62  2017         13        10

# TOP CRITICAL YEARS
Based on the critic top ten years, 1998 had the highest average critic score, with an impressive 9.32 across 10 games. This coincided with the year's best-selling game, Pokémon Red, developed by Game Freak and published by Nintendo.

Although 1998 achieved the highest average score, the number of games considered (10) was the second lowest among the top-ranked years

In [4]:
-- golden_years,
select public.critics_avg_year_rating.year,
	public.critics_avg_year_rating.num_games,
	public.critics_avg_year_rating.avg_critic_score,
	public.users_avg_year_rating.avg_user_score,
	(public.critics_avg_year_rating.avg_critic_score - public.users_avg_year_rating.avg_user_score) as diff
from public.users_avg_year_rating
join public.critics_avg_year_rating
on public.critics_avg_year_rating.year = public.users_avg_year_rating.year
where public.critics_avg_year_rating.avg_critic_score > 9 or public.users_avg_year_rating.avg_user_score > 9
order by public.critics_avg_year_rating.year

year  num_games  avg_critic_score  avg_user_score  diff
0  1997          8              7.93            9.50 -1.57
1  1998         10              9.32            9.40 -0.08
2  2004         11              9.03            8.55  0.48
3  2008         20              8.63            9.03 -0.40
4  2009         20              8.55            9.18 -0.63
5  2010         23              8.41            9.24 -0.83

# GOLDEN YEARS
In the Golden Years Results table, 1998 achieved the highest average critic score, while 1997 had the highest average user score. However, the difference between critic and user scores in 1997 was the largest, with a gap of 1.57 points—indicating a significant disparity in reception.

In contrast, 1998 had the smallest average score difference between critics and users, suggesting a strong alignment in critical and public opinion during that year.

